### ISSUE TITLE
Indication on whether a WSON node is 'Reconfigurable or not'

### ISSUE THREAD

**Issue by haomianzheng** [ietf-ccamp-wg/draft-ietf-ccamp-wson-yang#1]
This is related with another comments from ietf-layer0-types. 
https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-layer0-types/issues/6

According to the conclusion of this issue, the node type won't be specified in ietf-layer0-types (as they are WSON-specific, not L0-general). There will be a new leaf 'is-reconfigurable-node', in type boolean, in the WSON topology module to indicate whether the WSON node is reconfigurable or not. 

Setting to 1 means reconfigurable, as a ROADM; while setting to 0 means not reconfigurable, as a FOADM.

**Comment by haomianzheng** [comment]
Proposal as follow: 

OLD  
grouping wson-node-attributes {
    description "WSON node attributes";
    container wson-node {
      description "WSON node attrtibutes";
      leaf node-type {
        type identityref {
          base layer0-types:layer0-node-type;
        }
        description "WSON node type";
      }
    }
  }

NEW
  grouping wson-node-attributes {
    description "WSON node attributes";
    container wson-node {
      description "Candidate type: ROADM or FOADM";
      leaf is-reconfigurable-node {
        type boolean;
        description 
		  "true - ROADM, 
		   false - FOADM. ";
      }
    }
  }

**Comment by italobusi** [comment]
Some proposed changes:

1. Some editorial changes
2. make the definition more generic and applicable also to abstract TE nodes
3. add a default value
4. define the wson-node as a presence container: https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-layer0-types/issues/6#issuecomment-613875313

```
  grouping wson-node-attributes {
    description "WSON node attributes";
    container wson-node {
      presence "The TE node is a WSON node.";
      description "WSON node attributes";
      leaf is-reconfigurable-node {
        type boolean;
        default true;
        description
          "Indicates whether the WSON node is reconfigurable:
           - true: the node is reconfigurable
             (e.g., it is representing a ROADM node);
           - false: the node is not reconfigurable
             (e.g., it is representing a FOADM node).";
      }
    }
  }
```

**Comment by italobusi** [comment]
I think PR #3 is addressing this issue

I have linked this open issue with PR#3 so it will be automatically closed when the PR#3 is merged
